lib/spack/spack/package.py

```diff
@@ -956,7 +956,19 @@ def provides(self, vpkg_name):
 
     @property
     def installed(self):
-        return os.path.isdir(self.prefix)
+
+        has_prefix = os.path.isdir(self.prefix)
+        try:
+            # If the spec is in the DB, check the installed
+            # attribute of the record
+            rec = spack.store.db.get_record(self.spec)
+            db_says_installed = rec.installed
+        except KeyError:
+            # If the spec is not in the DB, the method
+            #  above raises a Key error
+            db_says_installed = False
+
+        return has_prefix and db_says_installed
```

Comment:
Here I did the simplest modification possible. Don't know if you want to raise, or log a message somewhere, in case we find something inconsistent, like `has_prefix == False` and `db_says_installed == True`.